plugins/techdocs-backend/src/service/router.ts

```diff
@@ -81,11 +83,27 @@ function isOutOfTheBoxOption(
 
 export async function createRouter(
   options: RouterOptions,
+  container: AwilixContainer,
 ): Promise<express.Router> {
   const router = Router();
-  const { publisher, config, logger, discovery } = options;
+  const { config, logger, discovery } = options;
   const catalogClient = new CatalogClient({ discoveryApi: discovery });
 
+  /**
+   * Resolving (instantiating) the actual implementation of 'techdocsGenerator' from the IoC container
+   *
+   * Notice stringly typed retrieval...
```

Comment:
Tbh I think it's always gonna be stringly typed in the end, just that the strings are wrapped up and delivered though constants or other containers. In the end each interface is gonna end up being identified by a single string